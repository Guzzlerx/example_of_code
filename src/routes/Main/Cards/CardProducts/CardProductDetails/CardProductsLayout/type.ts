import { CURRENCY_CODES_ENUM, PAYMENT_SYSTEMS_ENUM } from 'constants/cards/cards';
import { ICardProductAdvantagesProps } from '../CardProductDetails/Parts/CardProductAdvantages/type';

export interface ICardProductData {
  cardName: ICardProductAdvantagesProps['cardName'];
  currency: CURRENCY_CODES_ENUM;
  paymentSystem: PAYMENT_SYSTEMS_ENUM;
}

export interface IUseCardProductContext extends ICardProductData {
  showCardInfo: ({ cardName, currency, paymentSystem }: ICardProductData) => void;
}
