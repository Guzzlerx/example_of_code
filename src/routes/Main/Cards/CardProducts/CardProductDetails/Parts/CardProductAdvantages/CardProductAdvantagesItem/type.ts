import { CARD_NAMES_ENUM } from 'constants/cards/cards';

export interface ICardProductAdvantagesItemProps {
  cardName: Exclude<CARD_NAMES_ENUM, CARD_NAMES_ENUM.PERSONAL_LOAN>;
}
