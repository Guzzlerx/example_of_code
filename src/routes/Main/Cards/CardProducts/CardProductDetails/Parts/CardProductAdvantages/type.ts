import { CARD_NAMES_ENUM } from 'constants/cards/cards';
import { ICardConfigsAdvantages } from 'constants/cards/types';

export interface ICardProductAdvantagesProps {
  cardName: Exclude<CARD_NAMES_ENUM, CARD_NAMES_ENUM.PERSONAL_LOAN>;
  data: ICardConfigsAdvantages[];
}
