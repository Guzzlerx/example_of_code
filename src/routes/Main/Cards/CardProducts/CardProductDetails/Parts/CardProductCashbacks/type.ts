import { CARD_NAMES_ENUM } from 'constants/cards/cards';

export interface ICardProductCashbacksProps {
  cardName: Exclude<
    CARD_NAMES_ENUM,
    CARD_NAMES_ENUM.SMART_PAYOUT | CARD_NAMES_ENUM.INFINITY_PREMIUM | CARD_NAMES_ENUM.PERSONAL_LOAN
  >;
}
