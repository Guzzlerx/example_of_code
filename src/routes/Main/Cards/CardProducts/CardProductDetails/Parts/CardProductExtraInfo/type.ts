import { CARD_NAMES_ENUM } from 'constants/cards/cards';

export interface ICardProductExtraInfoProps {
  cardName: Extract<
    CARD_NAMES_ENUM,
    CARD_NAMES_ENUM.INFINITY_PREMIUM | CARD_NAMES_ENUM.SMART_PAYOUT
  >;
}
