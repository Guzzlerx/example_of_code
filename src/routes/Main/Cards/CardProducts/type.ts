import { CURRENCY_CODES_ENUM, PAYMENT_SYSTEMS_ENUM } from 'constants/cards/cards';
import { ICardProductAdvantagesProps } from './CardProductDetails/Parts/CardProductAdvantages/type';

export interface IMyCardProductItem {
  id: string;
  currencyCode: CURRENCY_CODES_ENUM;
  paymentSystem: PAYMENT_SYSTEMS_ENUM;
  cardName: ICardProductAdvantagesProps['cardName'];
}
