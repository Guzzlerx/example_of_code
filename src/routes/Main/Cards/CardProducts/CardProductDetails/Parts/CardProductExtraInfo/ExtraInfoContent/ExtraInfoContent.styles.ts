import styled from 'styled-components';
import { ATitle } from 'components/ATitle/ATitle';
import { COLOR_PALETTE } from 'constants/colors/colors';

const { gray700 } = COLOR_PALETTE;

export const StyledSubtitle = styled(ATitle)<{ isOneColumn: boolean }>`
  text-align: left;
  width: ${props => (props.isOneColumn ? '526px' : '493px')};
  margin: 24px 0 16px;
  color: ${gray700};
`;

export const StyledContainer = styled.div<{ isOneColumn: boolean }>`
  flex-direction: column;
  text-align: left;
  max-width: 526px;
  padding-left: ${props => (props.isOneColumn ? '24px' : '72px')};
`;

export const StyledUl = styled.ul`
  text-align: left;
  display: flex;
  flex-wrap: wrap;
  margin-left: 25px;
  color: ${gray700};
`;

export const StyledLi = styled.li<{ isOneColumn: boolean }>`
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  flex: ${props => (props.isOneColumn ? '1 1 100%' : '1 1 50%')};
  min-width: ${props => (props.isOneColumn ? '100%' : '50%')};
  margin-bottom: 8px;

  & strong {
    font-weight: 700;
  }
`;
