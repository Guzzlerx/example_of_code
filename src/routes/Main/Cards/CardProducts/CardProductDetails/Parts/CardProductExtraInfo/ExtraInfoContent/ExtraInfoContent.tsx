import { Trans, useTranslation } from 'react-i18next';
import { ATitle } from 'components/ATitle/ATitle';
import { CARD_NAMES_ENUM } from 'constants/cards/cards';
import { COLOR_PALETTE } from 'constants/colors/colors';
import { ICardProductExtraInfoProps } from '../type';
import { StyledContainer, StyledLi, StyledSubtitle, StyledUl } from './ExtraInfoContent.styles';

export const ExtraInfoContent = ({ cardName }: ICardProductExtraInfoProps) => {
  const { t } = useTranslation('cards');

  const { gray900 } = COLOR_PALETTE;

  const translations = {
    title: t(`card_product_extra_info.${cardName}.title`),
    subtitle: t(`card_product_extra_info.${cardName}.subtitle`),
  };

  const listItemPremium = [
    t(`card_product_extra_info.infinity_premium.list_item.1`),
    t(`card_product_extra_info.infinity_premium.list_item.2`),
    t(`card_product_extra_info.infinity_premium.list_item.3`),
    t(`card_product_extra_info.infinity_premium.list_item.4`),
    t(`card_product_extra_info.infinity_premium.list_item.5`),
    t(`card_product_extra_info.infinity_premium.list_item.6`),
    t(`card_product_extra_info.infinity_premium.list_item.7`),
    t(`card_product_extra_info.infinity_premium.list_item.8`),
  ];

  const listItemSmart = [
    t(`card_product_extra_info.smart_payout.list_item.1`),
    t(`card_product_extra_info.smart_payout.list_item.2`),
    t(`card_product_extra_info.smart_payout.list_item.3`),
  ];

  const firstColumnIndex = 4;
  const listItems = cardName === CARD_NAMES_ENUM.INFINITY_PREMIUM ? listItemPremium : listItemSmart;
  const firstColumnItems = listItems.slice(0, firstColumnIndex);
  const secondColumnItems = listItems.slice(firstColumnIndex);
  const isOneColumn = firstColumnItems.length <= 3;

  return (
    <StyledContainer isOneColumn={isOneColumn}>
      <ATitle size={20} weight={700} lineHeight={28} color={gray900}>
        {translations.title}
      </ATitle>
      <StyledSubtitle size={14} weight={400} lineHeight={20} isOneColumn={isOneColumn}>
        {translations.subtitle}
      </StyledSubtitle>
      <StyledUl>
        {firstColumnItems.map(item => (
          <StyledLi key={item} isOneColumn={isOneColumn}>
            <Trans>{item}</Trans>
          </StyledLi>
        ))}
        {secondColumnItems.map(item => (
          <StyledLi key={item} isOneColumn={isOneColumn}>
            {item}
          </StyledLi>
        ))}
      </StyledUl>
    </StyledContainer>
  );
};
