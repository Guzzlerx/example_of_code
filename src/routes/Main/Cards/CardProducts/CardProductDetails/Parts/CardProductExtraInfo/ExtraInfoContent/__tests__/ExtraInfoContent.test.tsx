import { RenderResult, render, screen } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { CARD_NAMES_ENUM } from 'constants/cards/cards';
import i18n from 'utils/languages/i18nForTests';
import { ICardProductExtraInfoProps } from '../../type';
import { ExtraInfoContent } from '../ExtraInfoContent';

describe('ExtraInfoContent', () => {
  let renderResult: RenderResult;

  const props: ICardProductExtraInfoProps = {
    cardName: CARD_NAMES_ENUM.SMART_PAYOUT,
  };

  beforeEach(() => {
    renderResult = render(
      <I18nextProvider i18n={i18n}>
        <ExtraInfoContent {...props} />
      </I18nextProvider>,
    );
  });

  it('should render correctly', () => {
    expect(renderResult.container).toBeInTheDocument();
  });

  it('snapshot', () => {
    expect(renderResult.container).toMatchSnapshot();
  });

  it('should render title correctly', () => {
    const element = screen.getByText(/Searching for a new way of getting money?/i);

    expect(element).toBeInTheDocument();
  });

  it('should render subtitle correctly', () => {
    const element = screen.getByText(
      /Thanks to a new program, provided by our bank, you have a possibility to open a deposit account in any type of crypto. This is what you get:/i,
    );

    expect(element).toBeInTheDocument();
  });

  it('should render listItems correctly', () => {
    const element = screen.getByText(
      /Diversification. Adding crypto savings to your portfolio can help mitigate risk;/i,
    );

    expect(element).toBeInTheDocument();
  });
});
