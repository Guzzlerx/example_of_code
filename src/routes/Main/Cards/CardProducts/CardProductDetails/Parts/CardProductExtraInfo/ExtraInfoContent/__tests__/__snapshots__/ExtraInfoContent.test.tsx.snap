// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtraInfoContent snapshot 1`] = `
<div>
  <div
    class="sc-dkrFOg hCbwQA"
  >
    <h2
      class="sc-bcXHqe ljoRHl"
      color="#212529"
      size="20"
    >
      Searching for a new way of getting money?
    </h2>
    <h2
      class="sc-bcXHqe fGtxeV sc-gswNZR gESawe"
      size="14"
    >
      Thanks to a new program, provided by our bank, you have a possibility to open a deposit account in any type of crypto. This is what you get:
    </h2>
    <ul
      class="sc-hLBbgP grRhQU"
    >
      <li
        class="sc-eDvSVe klQBpo"
      >
        Higher interest rate. Get up to 
        <strong>
          7% APR
        </strong>
        , 2 times more than with usual deposit;
      </li>
      <li
        class="sc-eDvSVe klQBpo"
      >
        Diversification. Adding crypto savings to your portfolio can help mitigate risk;
      </li>
      <li
        class="sc-eDvSVe klQBpo"
      >
        Security. Cryptocurrencies are stored on decentralized blockchain networks, which are highly secure and resistant to hacking.
      </li>
    </ul>
  </div>
</div>
`;
